### PR TITLE
Vcf split bcftools subworkflow

### PR DESCRIPTION
**Pull request by LouisLeNezet**
Add new subworkflow `vcf_split_bcftools`.
Update at the same time `bcftools_merge` modules.

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/bcftools/merge/main.nf [hunk 1]
```diff
@@ -8,13 +8,12 @@ process BCFTOOLS_MERGE {
         'biocontainers/bcftools:1.20--h8b25389_0' }"
 
     input:
-    tuple val(meta), path(vcfs), path(tbis)
-    tuple val(meta2), path(fasta)
-    tuple val(meta3), path(fai)
-    path(bed)
+    tuple val(meta), path(vcfs), path(tbis), path(bed)
+    tuple val(meta2), path(fasta), path(fai)
```
Comment (maxulysse) [inline modules/nf-core/bcftools/merge/main.nf:12]:
```suggestion
    tuple val(meta2), path(fasta)
    tuple val(meta2), path(fai)
```

I'd say this is more the nf-core way

Comment (LouisLeNezet) [reply modules/nf-core/bcftools/merge/main.nf]: Could you explain, why this is the case ?
In the previous it was even `tuple val(meta3), path(fai)` and I don't understand why we separate the fasta from it's index 🤔

File: modules/nf-core/bcftools/merge/tests/merge.config [hunk 2]
```diff
@@ -0,0 +1,8 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
```
Comment (SPPearce) [inline modules/nf-core/bcftools/merge/tests/merge.config:3]:
```suggestion
```
Don't think we need this bit any more.

File: modules/nf-core/bcftools/merge/main.nf [hunk 3]
```diff
@@ -8,13 +8,13 @@ process BCFTOOLS_MERGE {
         'biocontainers/bcftools:1.20--h8b25389_0' }"
 
     input:
-    tuple val(meta), path(vcfs), path(tbis)
+    tuple val(meta), path(vcfs), path(tbis), path(bed)
```
Comment (SPPearce) [inline modules/nf-core/bcftools/merge/main.nf:11]: Where is the bed file used?

File: modules/nf-core/bcftools/merge/meta.yml [hunk 4]
```diff
@@ -37,41 +41,24 @@ input:
...
-      type: file
-      description: VCF merged output file => when `--output-type v` is used
-      pattern: "*.vcf"
-  - bcf_gz:
+  - merged_variants:
       type: file
-      description: BCF merged output file (bgzipped) => when `--output-type b` is used
-      pattern: "*.bcf.gz"
-  - bcf:
+      description: Merged VCF output file it's format depends on the `--output-type` parameter
```
Comment (SPPearce) [inline modules/nf-core/bcftools/merge/meta.yml:56]:
```suggestion
      description: Merged VCF output file whose format depends on the `--output-type` parameter
```

**Pull request closed without merging**